core/controllers/base_test.py

```diff
@@ -288,6 +293,25 @@ def test_keys_match_en_qqq(self):
         qqq_key_list = self._extract_keys_from_json_file('qqq.json')
         self.assertEqual(en_key_list, qqq_key_list)
 
+    def test_keys_in_source_code_match_en(self):
+        """Tests that keys in HTML files are present in en.json."""
+        en_key_list = self._extract_keys_from_json_file('en.json')
+        for root, _, files in os.walk(os.path.join(
+                os.getcwd(), self.get_static_asset_filepath())):
+            for filename in files:
+                if filename.endswith('.html'):
+                    html_key_list = self._extract_keys_from_html_file(
+                        os.path.join(root, filename))
+                    self.assertLessEqual(set(html_key_list), set(en_key_list))
+                    if set(html_key_list) > set(en_key_list):
```

Comment:
Done. It is possible to pass a custom error message to the assert call, but it would face the same problem of ending the test right there, not going through the remaining files.